### DISCUSSION TITLE
How get string from MethodDefinition

### DISCUSSION THREAD

**Discussion by ttodua** [Washi1337/AsmResolver#341] [General]
### Description

Naive question, but as I'm unfamiliar, have to ask:   with `md.FullName` we can get method names as string. Is there any way to get  method body as string? i couldnt find which method/property does that.

**Comment by Washi1337** [comment]
There is nothing built-in to create a full string of a method body. However you can access all its components (instructions, locals, exception handlers etc) that you can manually format to your liking. There is also `CilInstructionFormatter` if you want some base for modifying the format of instructions.

**Reply by ttodua** [reply]
thanks for quick response. As I mentioned, I am not well-versed in low-level things like that, can you please give me a tiny example for that?
i.e. `Console.Write (methodDefinition.CilMethodBody.Instructions.ToString()` or ?

**Reply by Washi1337** [reply]
A very simple way to print all the instructions would be to go over all of them and just call `ToString`:

```csharp
for (var instruction in body.Instructions)
    Console.WriteLine(instruction.ToString());
```

A similar thing can be done for local variables:

```csharp
for (var local in body.LocalVariables)
    Console.WriteLine($"[{local.Index}] {local.VariableType}");
```

Exception handlers are slightly more complicated, as there are multiple types of handlers, but again a similar approach will do.

**Reply by ttodua** [reply]
thanks!
